sources/packages/backend/libs/test-utils/src/factories/application.ts

```diff
@@ -29,3 +48,98 @@ export function createFakeApplication(relations?: {
+  const applicationRepo = dataSource.getRepository(Application);
+  const studentAssessmentRepo = dataSource.getRepository(StudentAssessment);
+  const offeringRepo = dataSource.getRepository(EducationProgramOffering);
+  // Ensure student/user creation.
+  let savedUser: User;
+  let savedStudent: Student;
+  if (relations?.student) {
+    savedUser = relations.student.user;
+    savedStudent = relations.student;
+  } else {
+    savedUser = await userRepo.save(createFakeUser());
+    savedStudent = await studentRepo.save(createFakeStudent(savedUser));
+  }
+  // Create and save application.
+  const fakeApplication = createFakeApplication({ student: savedStudent });
+  fakeApplication.applicationStatus = ApplicationStatus.Enrolment;
+  const savedApplication = await applicationRepo.save(fakeApplication);
+  // Original assessment.
+  const fakeOriginalAssessment = createFakeStudentAssessment({
+    auditUser: savedUser,
+  });
+  fakeOriginalAssessment.application = savedApplication;
+  const disbursementSchedules: DisbursementSchedule[] = [];
+  // Original assessment - first disbursement.
+  const firstSchedule = createFakeDisbursementSchedule({
+    auditUser: savedUser,
+    disbursementValues: relations?.disbursementValues ?? [
+      createFakeDisbursementValue(DisbursementValueType.CanadaLoan, "CSLF", 1),
+    ],
+  });
+  firstSchedule.coeStatus = COEStatus.required;
+  firstSchedule.disbursementScheduleStatus = DisbursementScheduleStatus.Pending;
+  disbursementSchedules.push(firstSchedule);
+  if (options?.createSecondDisbursement) {
+    // Original assessment - second disbursement.
+    const secondSchedule = createFakeDisbursementSchedule({
+      auditUser: savedUser,
+      disbursementValues: relations?.disbursementValues ?? [
+        createFakeDisbursementValue(DisbursementValueType.BCLoan, "BCSL", 1),
+      ],
+    });
+    secondSchedule.coeStatus = COEStatus.required;
+    secondSchedule.disbursementScheduleStatus =
+      DisbursementScheduleStatus.Pending;
+    // First schedule is created with the current date as default.
+    // Adding 60 days to create some time between the first and second schedules.
+    secondSchedule.disbursementDate = getISODateOnlyString(addDays(60));
+    disbursementSchedules.push(secondSchedule);
+  }
+  fakeOriginalAssessment.disbursementSchedules = disbursementSchedules;
+  // Offering.
+  const fakeOffering = createFakeEducationProgramOffering({
+    institution: relations?.institution,
+    institutionLocation: relations?.institutionLocation,
+    auditUser: savedUser,
+  });
+  fakeOffering.offeringIntensity = OfferingIntensity.fullTime;
+  const savedOffering = await offeringRepo.save(fakeOffering);
+  fakeOriginalAssessment.offering = savedOffering;
+  const savedOriginalAssessment = await studentAssessmentRepo.save(
+    fakeOriginalAssessment,
+  );
+  savedApplication.currentAssessment = savedOriginalAssessment;
+  savedApplication.applicationStatus = ApplicationStatus.Enrolment;
```

Comment:
mm, we are already saving the status at line 95 right? then why this assignment

Reply:
I do not see a reason to. I will remove the last one.

Reply:
Removed.